src/main/java/edu/kit/compiler/parser/Parser.java

```diff
@@ -123,160 +193,201 @@ private void parseBlock() {
                 check(TokenType.Identifier, TokenType.Identifier) ||
                 check(TokenType.Identifier, TokenType.Operator_BracketL, TokenType.Operator_BracketR)) {
                 // LocalVariableDeclarationStatement
-                parseType();
-                expect(TokenType.Identifier);
+                Token firstToken = tokenStream.get();
+                DataType variableType = parseType();
+                Optional<ExpressionNode> expr = Optional.empty();
+                Token name = expect(TokenType.Identifier);
                 if (tokenStream.get().getType() == TokenType.Operator_Equal) {
                     tokenStream.next();
-                    parseExpression();
+                    expr = Optional.of(parseExpression());
                 }
                 expect(TokenType.Operator_Semicolon);
+                stmts.add(new LocalVariableDeclarationStatementNode(
+                    firstToken.getLine(), firstToken.getColumn(), variableType, name.getIntValue().get(), expr, false
+                ));
             } else {
                 // Statement
-                parseStatement();
+                Optional<StatementNode> stmt = parseStatement();
+                stmt.ifPresent(node -> stmts.add(node));
             }
         }
         expect(TokenType.Operator_BraceR);
+        return new BlockStatementNode(openingBrace.getLine(), openingBrace.getColumn(), stmts, false);
     }
 
-    private void parseStatement() {
+    private Optional<StatementNode> parseStatement() {
         Token token = tokenStream.get();
         switch(token.getType()) {
             case Operator_BraceL: {
-                parseBlock();
-                break;
+                return Optional.of(parseBlock());
             }
             case Operator_Semicolon: {
                 // EmptyStatement
                 tokenStream.next();
-                break;
+                return Optional.empty();
             }
             case Keyword_If: {
                 tokenStream.next();
                 expect(TokenType.Operator_ParenL);
-                parseExpression();
+                ExpressionNode condition = parseExpression();
                 expect(TokenType.Operator_ParenR);
-                parseStatement();
+                StatementNode thenStmt = parseStatement().orElse(emptyBlock(token));
+                Optional<StatementNode> elseStmt = Optional.empty();
                 if (tokenStream.get().getType() == TokenType.Keyword_Else) {
                     tokenStream.next();
-                    parseStatement();
+                    elseStmt = Optional.of(parseStatement().orElse(emptyBlock(token)));
```

Comment:
That's a lot of options around here. Would it be possible to simply return a `StatementNode` from `parseStatement` (with `emptyBlock` instead of `Optional.empty`? This way we would also keep empty blocks inside another block in the statement list, which could be required for the pretty printer(?).

Reply:
I did return a `StatementNode` at first, but the pretty print example in the assignment explicitly deletes empty statements, so I decided to implement the same behavior. It was also mentioned in the meeting that we can delete empty statements.

Reply:
I see. You could catch empty statements in the `parseBlock` method like local variable declarations. But I don't know if that's more readable.

Reply:
Good idea, I think this improves it a bit